src/Extension/FluentExtension.php

```diff
@@ -1176,4 +1197,110 @@ public function updateLocalisationTabConfig(GridFieldConfig $config)
             }
         }
     }
+
+    public function getLocalisedCopyActive(): bool
+    {
+        return $this->localisedCopyActive;
+    }
+
+    public function setLocalisedCopyActive(bool $active): DataObject
+    {
+        $this->localisedCopyActive = $active;
+
+        return $this->owner;
+    }
+
+    /**
+     * Localised copy global state manipulation
+     * useful for disabling localised copy feature in parts of the code
+     *
+     * @param callable $callback
+     * @return mixed
+     */
+    public function withLocalisedCopyState(callable $callback)
+    {
+        $active = $this->localisedCopyActive;
+
+        try {
+            return $callback();
+        } finally {
+            $this->localisedCopyActive = $active;
+        }
+    }
+
+    /**
+     * Duplicate related objects based on configuration
+     * Provides an extension hook for custom duplication
+     */
+    protected function makeLocalisedCopy(): void
+    {
+        if (!$this->localisedCopyNeeded()) {
+            return;
+        }
+
+        $owner = $this->owner;
+        $relations = (array) $owner->config()->get('localised_copy');
+
+        $owner->invokeWithExtensions('onBeforeLocalisedCopy');
+
+        foreach ($relations as $relation => $field) {
+            $original = $owner->{$relation}();
+
+            if (!$original instanceof DataObject) {
+                continue;
+            }
+
+            if (!$original->exists()) {
+                continue;
+            }
+
+            $duplicate = $original->duplicate();
+            $owner->{$field} = $duplicate->ID;
```

Comment:
Be careful, polymorphic relations (has_one -> dataobject) have both a RelationID and RelationClass field.

I may suggest that instead of localised_copy declaring both the relation and the field, that you make it a flat list of fields. Then instead of using `{$field}` you can just use `setComponent($relation, $duplicate)` directly. dataobject will handle all the relation polymorphism for you.

Reply:
Good catch, will try it out :)

Reply:
Works like a charm :D